src/funcx_common/redis/connection.py

```diff
@@ -10,55 +10,62 @@
     has_redis = False
 
 log = logging.getLogger(__name__)
-
 RT = t.TypeVar("RT")
 
+_CONNECTION_FACTORY_T = t.Callable[[str, int], "redis.Redis"]
+_OPT_CONNECTION_FACTORY_T = t.Optional[_CONNECTION_FACTORY_T]
+
+
+def default_redis_connection_factory(hostname: str, port: int) -> "redis.Redis":
+    # defer errors over redis being present/absent until someone actually
+    # tries to construct a connection
+    # this allows general use of `funcx-common` without `redis` installed
+    if not has_redis:
+        raise RuntimeError(
+            "Cannot construct a redis connection if the 'redis' "
+            "package is not available. Either install it explicitly or install the "
+            "'redis' extra, as in\n"
+            "  pip install 'funcx-common[redis]'"
+        )
+
+    return redis.Redis(
+        host=hostname,
+        port=port,
+        decode_responses=True,
+        health_check_interval=30,
+    )
 
-class FuncxRedisConnection:
+
+class HasRedisConnection:
     """
-    A basic redis client wrapper which can be used to inherit and provide a
+    A redis client wrapper which can be used to inherit and provide a
     working `.redis_client` attribute to child classes.
     """
 
-    def __init__(self, hostname: str, *, port: int = 6379):
-        # defer errors over redis being present/absent until someone actually
-        # tries to construct a task queue
-        # at that point, if redis is not present, error
-        # this makes all code in this class which uses `redis` unreachable in
-        # that case *except* for method type annotations (for which reason we will
-        # quote 'redis' in type annotations to avoid runtime evaluation)
-        if not has_redis:
-            raise RuntimeError(
-                "Cannot construct a FuncxRedisTaskQueue if the 'redis' package "
-                "is not available. Either install it explicitly or install the "
-                "'redis' extra, as in\n"
-                "  pip install 'funcx-common[redis]'"
-            )
-
-        self.hostname = hostname
-        self.port = port
-        self.redis_client = redis.Redis(
-            host=self.hostname, port=self.port, decode_responses=True
+    def __init__(
+        self,
+        hostname: str,
+        *,
+        port: int = 6379,
+        redis_connection_factory: _OPT_CONNECTION_FACTORY_T = None,
+    ) -> None:
+        redis_connection_factory = (
+            redis_connection_factory or default_redis_connection_factory
         )
+        self.redis_client = redis_connection_factory(hostname, port)
 
     def _get_str_attrs(self) -> t.List[str]:
-        return [f"hostname={self.hostname}", f"port={self.port}"]
+        return [str(self.redis_client)]
```

Comment:
repr of redis_client vs str? this is (only?) used in __repr__

Reply:
I'm not sure, are you asking why I used `str()` instead of `repr()` here? I probably should use `repr()` (though they are the same in this case).

I'll switch it, but I kind of want to remove more of this stuff... I think inheriting from a connection-providing-class is a bad pattern.